project/src/main/java/com/google/sps/data/PlacesScorer.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.data;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import com.google.maps.model.LatLng;
+
+public class PlacesScorer {
+
+    private static final double RATING_WEIGHT = 0.7;
+    private static final double DURATION_WEIGHT = 0.3;
+    private static final double MAX_RATING = 5;
+    private static final double MAX_DURATION_SECONDS = 40 * 60;
+    private ImmutableMap<Place, Double> durations;
+    private ImmutableList<Place> places;
+
+    /**
+     * Constructor for PlacesScorer, calculates the durations map used for scoring.
+     *
+     * @param placesToScore A list of places we want to calculate their score
+     * @param userLocation The user's physical location used for duration calculations
+     */
+    public PlacesScorer(ImmutableList<Place> placesToScore, LatLng userLocation) {
```

Comment:
Because than I will just need to pass them between all the methods instead of having them as fields and available for all methods. Also, regrading our discussion today I think it makes sense for the durations map to be a field of a PlacesScorer instance because it defines the scores this instance will generate.

Reply:
I decided to change it to be in the `getScores` methos since it is easier for testing and also to take the "lazy" approach.